<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://www.springframework.org/schema/beans"
	xmlns:ws="http://jax-ws.dev.java.net/spring/core"
	xmlns:wss="http://jax-ws.dev.java.net/spring/servlet"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:aop="http://www.springframework.org/schema/aop"
	xmlns:p="http://www.springframework.org/schema/p"
	xmlns:context="http://www.springframework.org/schema/context"
	xmlns:jee="http://www.springframework.org/schema/jee"
	xmlns:tx="http://www.springframework.org/schema/tx"
	xsi:schemaLocation="
          http://www.springframework.org/schema/aop http://www.springframework.org/schema/aop/spring-aop-2.5.xsd
		  http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-2.5.xsd 
          http://jax-ws.dev.java.net/spring/core http://jax-ws.dev.java.net/spring/core.xsd 
          http://jax-ws.dev.java.net/spring/servlet http://jax-ws.dev.java.net/spring/servlet.xsd
          http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context-2.5.xsd
    	  http://www.springframework.org/schema/jee http://www.springframework.org/schema/jee/spring-jee-2.5.xsd
		  http://www.springframework.org/schema/tx http://www.springframework.org/schema/tx/spring-tx-2.5.xsd">

	<bean id="mapExceptionAdvisor"
		class="org.kuali.student.poc.common.util.ExceptionMappingAdvice">
		<property name="exceptionMapping">
			<map>
				<entry key="javax.persistence.EntityExistsException"
					value="org.kuali.student.poc.common.ws.exceptions.AlreadyExistsException" />
			</map>
		</property>
	</bean>
	<aop:config>
		<aop:aspect id="dataAccessToBusinessException"
			ref="mapExceptionAdvisor">

			<aop:after-throwing
				pointcut="execution(* org.kuali.student.poc.personidentity.person.service.*.*(..))"
				method="afterThrowing" throwing="ex" />

		</aop:aspect>
	</aop:config>
</beans>